src/DGtal/geometry/curves/ArithmeticalDSLKernel.ih

```diff
@@ -257,6 +257,113 @@ DGtal::ArithmeticalDSLKernel<TCoordinate,adjacency>
     }
 }
 
+
+// ----------------------------------------------------------------------------
+template <typename TCoordinate, unsigned short adjacency>
+template <typename TInteger>
+inline
+typename DGtal::ArithmeticalDSLKernel<TCoordinate,adjacency>::Octant
+DGtal::ArithmeticalDSLKernel<TCoordinate,adjacency>
+::octant(const TInteger& a, const TInteger& b) 
```

Comment:
ArithmeticalDSLKernel aims at gathering and specializing small functions that depends on thickness (naive or standard). However `octant` does not depends on thickness. Therefore it should be defined in ArithmeticalDSL instead of ArithmeticalDSLKernel where the code is duplicated in the naive and in the standard case.